internal/storage/factory.go

```diff
@@ -3,23 +3,39 @@ package storage
 import (
 	"context"
 	"errors"
+
+	"github.com/milvus-io/milvus/internal/util/paramtable"
 )
 
 type ChunkManagerFactory struct {
-	cacheStorage  string
-	vectorStorage string
-	config        *config
+	persistentStorage string
+	config            *config
+}
+
+func NewChunkManagerFactoryWithParam(params *paramtable.ComponentParam) *ChunkManagerFactory {
+	if params.CommonCfg.StorageType == "local" {
+		return NewChunkManagerFactory("local", RootPath(params.LocalStorageCfg.Path))
+	}
+	return NewChunkManagerFactory("minio",
```

Comment:
It seems the storage type does not check the availability before here, so it should raise an error if storage type is not supported.

Reply:
error happens when you do NewPersistentStorageChunkManager.
We didn't handle error when you create factory